bundles/org.openhab.core.transform/src/main/java/org/openhab/core/transform/TransformationHelper.java

```diff
@@ -172,4 +173,56 @@ public static boolean isTransform(String pattern) {
             throw new TransformationException("Transformation service threw an exception: " + e.getMessage(), e);
         }
     }
+
+    /**
+     * Transforms a state using transformation functions within a given pattern.
+     *
+     * @param transformationString the pattern that contains the transformation instructions
+     * @param inputFormat the pattern to format the value being passed into the transformation, <code>null</code> to use
+     *            the one extracted from parameter transformationString
+     * @param state the state to be formatted before being passed into the transformation function
+     * @return the result of the transformation. If no transformation was done, <code>null</code> is returned
+     * @throws TransformationException if transformation service is not available or the transformation failed
+     */
+    public static @Nullable String transform(String transformationString, @Nullable String inputFormat, State state)
```

Comment:
I don't think this should be part of the `TransformationHelper`. It is very specific to UI/SSE needs and should reside there.

Reply:
I need it in two places in the core framework.

At least the second method I added will be necessary in the transformation addons like for example here:
https://github.com/openhab/openhab-addons/blob/cc439e2b170600d9e5996b8c9a1219e7aecae2f5/bundles/org.openhab.transform.map/src/main/java/org/openhab/transform/map/internal/profiles/MapTransformationProfile.java#L119

Reply:
For this particular method, I can replicate the code in the two places it is used and remove it from Transformation Helper.

Reply:
Method removed

Reply:
I finally removed the two methods I previously added.